src/components/classrooms/StudentsTable.js

```diff
@@ -0,0 +1,352 @@
+    this.setState({ page });
+  };
+
+  handleChangeRowsPerPage = event => {
+    const rowsPerPage = parseInt(event.target.value, 10);
+    this.setState({ page: 0, rowsPerPage });
+  };
+
+  handleChangeDense = event => {
+    this.setState({ dense: event.target.checked });
+  };
+
+  handleDeleteUser = row => {
+    const { id: userId } = row;
+    const {
+      dispatchDeleteUserInClassroom,
+      classroom,
+      userId: teacherId,
+    } = this.props;
+    dispatchDeleteUserInClassroom({
+      username: row[USERNAME_HEAD_CELL_ID],
+      userId,
+      teacherId,
+      classroomId: classroom.get('id'),
+    });
+  };
+
+  renderDenseSwitch = () => {
+    const { t } = this.props;
+    const { dense } = this.state;
+    const switchEl = (
+      <Switch
+        checked={dense}
+        color="primary"
+        onChange={this.handleChangeDense}
+      />
+    );
+    return <FormControlLabel control={switchEl} label={t('Dense padding')} />;
+  };
+
+  renderDeleteUserButton = row => {
+    const { t } = this.props;
+    return (
+      <Tooltip title={t('Delete this user.')}>
+        <IconButton
+          className={DELETE_USER_IN_CLASSROOM_BUTTON_CLASS}
+          color="inherit"
+          onClick={() => this.handleDeleteUser(row)}
+        >
+          <DeleteIcon />
+        </IconButton>
+      </Tooltip>
+    );
+  };
+
+  renderUserOperations = row => {
+    const { classroom } = this.props;
+    return (
+      <TableCell>
+        {this.renderDeleteUserButton(row)}
+        <EditUserInClassroomButton
+          classroomId={classroom.get('id')}
+          user={row}
+        />
+      </TableCell>
+    );
+  };
+
+  render() {
+    const {
+      selected,
+      page,
+      rowsPerPage,
+      dense,
+      order,
+      orderBy,
+      rows,
+    } = this.state;
+
+    const { classes, classroom } = this.props;
+
+    const isSelected = name => selected.indexOf(name) !== -1;
+
+    const headCells = this.buildHeadCells();
+
+    const emptyRows =
+      rowsPerPage - Math.min(rowsPerPage, rows.length - page * rowsPerPage);
+
+    return (
+      <div className={classes.root}>
+        <Paper className={classes.paper}>
+          <TableToolbar
+            numSelected={selected.length}
+            classroomName={classroom.get('name')}
+          />
+          <TableContainer>
+            <Table size={dense ? 'small' : 'medium'}>
+              <TableHeader
+                headCells={headCells}
+                classes={classes}
+                numSelected={selected.length}
+                order={order}
+                orderBy={orderBy}
+                onSelectAllClick={this.handleSelectAllClick}
+                onRequestSort={this.handleRequestSort}
+                rowCount={rows.length}
+              />
+              <TableBody id={CLASSROOM_TABLE_BODY_ID}>
+                {stableSort(rows, getComparator(order, orderBy))
+                  .slice(page * rowsPerPage, page * rowsPerPage + rowsPerPage)
```

Comment:
Maybe factor out these values with constant names that make their roles clearer?